File: api-java-ecommerce/src/main/java/busness/ecommerce/entity/UserProfile.java
package busness.ecommerce.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String firstName;
    private String lastName;
    private String phone;
    private String address;
    private String country;
    private String city;
    private String borough;
    private String photo;

    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;
}

File: lisungi/src/app/core/models/UserProfile.ts
export interface UserProfile{
  publicId?: string;
  firstName: string;
  lastName: string;
  gender?: string;
  phone: string;
  address: string;
  country: string;
  city: string;
  borough: string;
  profession?: string;
  photo: string;
  registrationDate?: Date;
  reasonRemoval?: string;
  userId: number;
  userName: string;
  userEmail: string;
}
